entity-framework/core/what-is-new/ef-core-9.0/breaking-changes.md

```diff
@@ -249,6 +250,38 @@ Doing this for all your top-level entity types will make EF behave just like bef
 
 At this point, if you wish, you can also update all your documents to rewrite their JSON `id` property. Note that this is only possible if entities of different types don't share the same id value within the same container.
 
+<a name="cosmos-key-changes"></a>
+
+#### The JSON `id` property is mapped to the key
+
+[Tracking Issue #34179](https://github.com/dotnet/efcore/issues/34179)
+
+##### Old behavior
+
+Previously, EF created a shadow property mapped to the JSON `id` property, unless one of the properties was mapped to `id` explicitly.
+
+##### New behavior
+
+Starting with EF Core 9, the key property will be mapped to the JSON `id` property by convention if possible.
```

Comment:
@AndriySvyryd reading this, I don't think it's clear what the actual functional impact is on users... Is it possible to include an example of what's actually breaking?